### PR TITLE
Only build when code files change. Resolves #11

### LINKED ISSUE #11: BuildBot: prevent builds for docs/non-code-only pull-requests/merges
Poor machines :cry: (I noticed that this wasn't addressed in https://github.com/monero-project/kovri/pull/452).

This includes markdown and things unrelated to compiling. I believe @danrmiller said this was a WIP. Ticketing for housekeeping.

### PR DESCRIPTION
**Pull request by danrmiller**
I will add this for monero too.

### CODE REVIEW

**Comment by anonimal** [comment]
Can we do `.conf` too?

File: buildbot/master/master.cfg [hunk 1]
```diff
@@ -64,6 +64,24 @@ c['change_source'].append(changes.GitPoller(
...
+    '''Determine whether we want to start a compile pass based on the files that
+        have changed.  Only skip compile if ALL changes are in the ignore criteria.
+    '''
+    compile = True
+    for fn in change.files:
+         # ignore all changes to gpg keys and markdown and text files
+         if fn.endswith('.asc') or fn.endswith('.md') or fn.endswith('.txt'):
+                compile = False
+         # don't ignore changes to CMakeLists.txt files
+         elif re.match('CMakeLists.txt', fn):
```
Comment (Jaqueeee) [inline buildbot/master/master.cfg]: shouldn't this be if instead of elif? CmakeLists.txt is matched on line 76.

Comment (danrmiller) [inline buildbot/master/master.cfg]: Thanks!

**Comment by danrmiller** [comment]
.conf added

**Comment by luigi1111** [comment]
Ready to merge?

**Comment by danrmiller** [comment]
@luigi1111 yes

**Pull request merged**